src/VisualStudio/Core/Def/Implementation/EditAndContinue/VisualStudioActiveStatementProvider.cs

```diff
@@ -0,0 +1,118 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.EditAndContinue;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.VisualStudio.Debugger;
+using Microsoft.VisualStudio.Debugger.Clr;
+using Microsoft.VisualStudio.Debugger.Symbols;
+using Roslyn.Utilities;
+
+namespace Microsoft.VisualStudio.LanguageServices.EditAndContinue
+{
+    [Export(typeof(IActiveStatementProvider)), Shared]
+    internal sealed partial class VisualStudioActiveStatementProvider : IActiveStatementProvider
+    {
+        public Task<ImmutableArray<ActiveStatementDebugInfo>> GetActiveStatementsAsync(CancellationToken cancellationToken)
+        {
+            DkmComponentManager.InitializeThread(DebuggerComponentIds.ManagedEditAndContinueService);
+            try
+            {
+                // TODO: report errors
```

Comment:
This seems important?

Reply:
Yeah, we swallow errors for now. I have a workitem tracking it.